presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardPredicate.java

```diff
@@ -108,54 +108,59 @@ public static ShardPredicate create(TupleDomain<RaptorColumnHandle> tupleDomain,
                 continue;
             }
 
+            StringJoiner columnPredicate = new StringJoiner(" OR ", "(", ")").setEmptyValue("true");
             Ranges ranges = domain.getValues().getRanges();
 
-            // TODO: support multiple ranges
-            if (ranges.getRangeCount() != 1) {
+            // prevent generating complicated metadata queries
+            if (ranges.getRangeCount() > MAX_RANGE_COUNT) {
                 continue;
             }
-            Range range = getOnlyElement(ranges.getOrderedRanges());
 
-            Object minValue = null;
-            Object maxValue = null;
-            if (range.isSingleValue()) {
-                minValue = range.getSingleValue();
-                maxValue = range.getSingleValue();
-            }
-            else {
-                if (!range.getLow().isLowerUnbounded()) {
-                    minValue = range.getLow().getValue();
+            for (Range range : ranges.getOrderedRanges()) {
+                StringJoiner rangePredicate = new StringJoiner(" AND ").setEmptyValue("true");
```

Comment:
Move this below, right before where it is needed.

Reply:
Add parenthesis around the `AND` range, so the precedence with `OR` is explicit.